### PR TITLE
Add Delete of issue support;

### PR DESCRIPTION
**Pull request by lukapr**
Add Deleted status to DB;
Add new REST-end point to be able to get issues by the statuses (+sorted by the requested field);

The delete functionality will work the same as other status changes -> using patch end point.

### CODE REVIEW

File: src/main/java/com/nepoladka/nepoladka/rest/IssueController.java [hunk 1]
```diff
@@ -58,6 +58,14 @@ public void setNotificationService(NotificationService notificationService) {
         return issueService.sortLatestOnTop(issueService.findAll());
     }
 
+    @GetMapping(value = {"/list", "/list/"})
```
Comment (romos) [inline src/main/java/com/nepoladka/nepoladka/rest/IssueController.java]: Not sure about what the best practice is, but can we somehow consolidate the base issue endpoint (`/`) and `/list`?
I mean they are pretty similar, but the latter is just with additional condition and sorting.

**First option** is to define everything in a single method like this:
```
public @ResponseBody List<Issue> getIssues(
    @RequestParam(name = "statuses", required = false) List<String> statuses,
    @RequestParam(name = "sortBy", required = false) String sortBy
    // more parameters to be added in future
) {
    // here we could decide if the optional parameters have been passed and we need to call a particular method of Repository
}
```

**Another way** is to keep both methods but map them to the same endpoint although with different parameters using `@RequestParam( param= { ... })`
```
@GetMapping(value = {"", "/"},
    params = {})
public @ResponseBody List<Issue> getIssues() { }
@GetMapping(value = {"", "/"},
    params = {statuses})
public @ResponseBody List<Issue> getIssues(
    @RequestParam(name = "statuses") List<String> statuses,
    @RequestParam(name = "sortBy", required = false) String sortBy
    //...
) {}
```
This option is slightly ambiguous, IMHO.

@lukapr , @utkillr, what do you think?

Comment (lukapr) [reply src/main/java/com/nepoladka/nepoladka/rest/IssueController.java]: The second option is not valid. It is not possible to have two methods with the same end point.
I agree with option 1. Will make the changes

File: src/main/java/com/nepoladka/nepoladka/repositories/IssueRepository.java [hunk 2]
```diff
@@ -15,4 +17,7 @@
     List<Issue> findAllByItem(Item item);
     List<Issue> findAllByItemAndProblem(Item item, Problem problem);
     Integer countAllByProblemAndStatusNotAndItem(Problem problem, IssueStatus status, Item item);
+
+    @Query("select i from Issue i where status.name in (:statuses)")
```
[the remark this answers is not in the document]
Comment (lukapr) [reply src/main/java/com/nepoladka/nepoladka/repositories/IssueRepository.java]: Sure. Will do

File: src/main/java/com/nepoladka/nepoladka/services/IssueService.java [hunk 3]
```diff
@@ -31,6 +32,10 @@ public void setIssueStatusService(IssueStatusService issueStatusService) {
         return issueRepository.findAll();
     }
 
+    public List<Issue> findAllByStatuses(List<String> statuses, String sortBy) {
```
[the remark this answers is not in the document]
Comment (lukapr) [reply src/main/java/com/nepoladka/nepoladka/services/IssueService.java]: Yeah, agree with you. Will think about this 2 approaches and make the changes

File: src/main/java/com/nepoladka/nepoladka/rest/IssueController.java [hunk 4]
```diff
@@ -54,18 +56,17 @@ public void setNotificationService(NotificationService notificationService) {
...
-                                                         @RequestParam(name = "sortBy") String sortBy) {
-        return issueService.findAllByStatuses(statuses, sortBy);
+    public @ResponseBody
+    List<Issue> getIssues(@RequestParam(name = "statuses", required = false) List<String> statuses,
+                          @RequestParam(name = "sortBy", required = false) String sortBy,
+                          @RequestParam(name = "sortDirection", defaultValue = "DESC") String sortDirection) {
+        if (!CollectionUtils.isEmpty(statuses) && sortBy != null) {
+            return issueService.findAllByStatusesSorted(statuses, sortBy, Sort.Direction.valueOf(sortDirection));
+        } else {
+            return issueService.sortLatestOnTop(issueService.findAll());
```
Comment (romos) [inline src/main/java/com/nepoladka/nepoladka/rest/IssueController.java]: Just one more comment here - and then let's merge.

Can we please get rid of this `sortLatestOnTop` method and replace it with smth like calling the same `findAllByStatusesSorted` but with some default parameters (e.g. statuses = [all statuses, except Deleted], sorting by create date, descending).

The code can be as follows (approximately):
... getIssues(...
) {
    ... statuses = [list of standard statuses to display]
    sorting = <some default sorting like by CreateDate, descending

    if (! CollectionUtils.isEmpty(statuses)....)
        // override with the externally provided params

    // single call to issueService.findAllByStatusesSorted(...)
}


The `sortLatestOnTop` was erroneously introduced earlier in https://github.com/romos/nepoladka/pull/16, but we decided to use standard Spring data approaches.

**Pull request merged**